### PR TITLE
translation: Update chapter_computational_complexity/iteration_and_recursion.md

### PR DESCRIPTION
**Pull request by Phoenix0415**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [ ] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [ ] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

**Comment by krahets** [comment]
Happy Lunar New Year!

Randomly assign ['@QiLOL' '@yuelinxin'] to review the pull request #1074

File: docs-en/chapter_computational_complexity/iteration_and_recursion.md [hunk 1]
```diff
@@ -20,21 +20,21 @@ The flowchart below represents this sum function.
...
 
 ```src
 [file]{iteration}-[class]{}-[func]{while_loop}
 ```
 
-**The `while` loop is more flexible than the `for` loop**. In a `while` loop, we can freely design the initialization and update steps of the condition variable.
+**`While` loops provide more flexibility than `for` loops**, especially since they allow for custom initialization and modification of the condition variable at each step.
 
-For example, in the following code, the condition variable $i$ is updated twice in each round, which would be inconvenient to implement with a `for` loop:
+For example, in the following code, the condition variable $i$ is updated twice in each round, which would be inconvenient to implement with a `for` loop.
```
Comment (QiLOL) [inline docs-en/chapter_computational_complexity/iteration_and_recursion.md:37]:
```diff
- twice in each round
+ twice each round
```

File: docs-en/chapter_computational_complexity/iteration_and_recursion.md [hunk 2]
```diff
@@ -106,10 +106,10 @@ In practice, the depth of recursion allowed by programming languages is usually
...
+Interestingly, **if a function performs its recursive call as the very last step before returning,** it can be optimized by the compiler or interpreter to be as space-efficient as iteration. This scenario is known as "tail recursion."
 
-- **Regular Recursion**: The function needs to perform more code after returning to the previous level, so the system needs to save the context of the previous call.
-- **Tail Recursion**: The recursive call is the last operation before the function returns, meaning no further actions are required upon returning to the previous level, so the system doesn't need to save the context of the previous level's function.
+- **Regular Recursion**: In standard recursion, when the function returns to the previous level, it continues to execute more code, requiring the system to save the context of the previous call.
+- **Tail Recursion**: Here, the recursive call is the final operation before the function returns. This means that upon returning to the previous level, no further actions are needed, so the system doesn't need to save the context of the previous level.
```
Comment (QiLOL) [inline docs-en/chapter_computational_complexity/iteration_and_recursion.md:112]:
```diff
- doesn't
+ does not
```

File: docs-en/chapter_computational_complexity/iteration_and_recursion.md [hunk 3]
```diff
@@ -191,4 +191,4 @@ Observing the above code, when recursion is transformed into iteration, the code
 - The transformed code may become harder to understand and less readable.
```
Comment (QiLOL) [inline docs-en/chapter_computational_complexity/iteration_and_recursion.md:191]:
```diff
- harder
+ more challenging
```

File: docs-en/chapter_computational_complexity/iteration_and_recursion.md [hunk 4]
```diff
@@ -20,21 +20,21 @@ The flowchart below represents this sum function.
 
 ![Flowchart of the Sum Function](iteration_and_recursion.assets/iteration.png)
 
-The number of operations in this sum function is proportional to the input data size $n$, or in other words, it has a "linear relationship". This is actually what **time complexity describes**. This topic will be detailed in the next section.
+The number of operations in this summation function is proportional to the size of the input data $n$, or in other words, it has a "linear relationship." In fact, this "linear relationship" is what time complexity describes. This topic will be discussed in more detail in the next section.
```
Comment (QiLOL) [inline docs-en/chapter_computational_complexity/iteration_and_recursion.md:23]: "In fact" may not be necessary.

**Review by QiLOL — COMMENTED** [review]
Thanks for the contribution, please check the comment(which is fully optional)! : )

**Review by krahets — APPROVED** [review]
Thanks for the PR @Phoenix0415 and review @QiLOL !

**Pull request merged**